src/v/datalake/serde_parquet_writer.cc

```diff
@@ -37,7 +37,10 @@ ss::future<writer_error> serde_parquet_writer::finish() {
 ss::future<std::unique_ptr<parquet_ostream>>
 serde_parquet_writer_factory::create_writer(
   const iceberg::struct_type& schema, ss::output_stream<char> out) {
-    serde::parquet::writer::options opts{.schema = schema_to_parquet(schema)};
+    serde::parquet::writer::options opts{
+      .schema = schema_to_parquet(schema),
+      .compress = true,
```

Comment:
I'm fine in general with the idea of compressing by default, but especially if we haven't benchmarked this yet, WDYT about allowing folks to opt out at the topic level? Or cluster level?

Reply:
@mattschumpert had asked for compression controls, which I guess I'm OK with as a tunable, but I don't want to support arbitrary compression codecs (especially the ones that are not futurized in the broker).

Reply:
@rockwotj zstd by default is great. agree we don't want open season , but Redpanda today does support 4 specific options on the Producer-Broker path at least: https://docs.redpanda.com/current/develop/produce-data/configure-producers/#compression-type. Ideally the topic would have a tunable for codecs we know will always be available on all our platforms (and which are in that list).   We dont want to spend a ton of time on this (beyond zstd always-on), but it would be important to maybe at least have an opt-out as @andrwng  says.  'Type=none' or something like that

Reply:
For highly CPU bound systems I could see the need to (perhaps as an escape hatch when in trouble or perhaps as an opinionated conf) not to compress e.g. messages that may themselves already be quite compressed.

Reply:
I will followup with a cluster configuration to disable compression, but it will be extendable in the future to different codecs, but for GA we will only support zstd

Reply:
which zstd compression level is this?

Reply:
The default, which is 3 IIRC?